spec/models/order_spec.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: orders
+#
+#  id            :bigint           not null, primary key
+#  restaurant_id :bigint
+#  order_id      :string
+#  customer_name :string
+#  time          :datetime
+#  publish_time  :datetime
+#  status        :string
+#  info          :json
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  context 'validations' do
+
+    it { should validate_presence_of(:order_id) }
+    it { should validate_presence_of(:customer_name) }
+    it { should validate_presence_of(:time) }
+    it { should validate_presence_of(:publish_time) }
+    it { should validate_presence_of(:status) }
+    
+    it { should validate_uniqueness_of(:order_id) }
+    
+    it { should belong_to(:restaurant) } 
+
+    
+    it 'ensure status valid value' do
+      order = Order.create(order_id: 'order_uuid', customer_name: 'name dummy', time: '2019-08-06 18:29:57', publish_time: '2019-08-06 18:29:57', status: 'DummyStatus')
```

Comment:
1. Create a factory for order that will automatically generate dummy data
2. Use `build` instead of `create` - we don't need to persist to the DB to check validity (I think)
3. Just so you know, when there's a boolean method ending with a question mark, rspec has magic syntax:
`expect(order).to_not be_valid`